lib/teleterm/services/connectmycomputer/connectmycomputer.go

```diff
@@ -254,3 +257,49 @@ func (c *RoleSetupConfig) CheckAndSetDefaults() error {
 
 	return nil
 }
+
+type TokenProvisioner struct {
+	Log *logrus.Entry
+}
+
+// CreateNodeToken creates a node join token that is valid for 5 minutes.
+func (t *TokenProvisioner) CreateNodeToken(ctx context.Context, provisioner Provisioner, cluster *clusters.Cluster) (types.ProvisionToken, error) {
+	tokenName, err := utils.CryptoRandomHex(auth.TokenLenBytes)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var req types.ProvisionTokenSpecV2
+	req.Roles = types.SystemRoles{types.RoleNode}
+	req.SuggestedLabels = types.Labels{
+		types.ConnectMyComputerNodeOwnerLabel: apiutils.Strings{cluster.GetLoggedInUser().Name},
+	}
```

Comment:
>I'd rather avoid other devs thinking that we actually need SuggestedLabels when it appears to be more of a convenient transport method than something that we actually need.

100% agree, I went for it because it was a "convenient transport method". We don't actually use it, so returning our own struct with only two fields makes sense to me. Done.